ga/server/periodic/ga-server-periodic.cpp

```diff
@@ -205,9 +324,76 @@ handle_netreport(ctrlmsg_system_t *msg) {
 	return;
 }
 
+void
+handle_reconfig(ctrlmsg_system_t *msg){
+	ctrlmsg_system_reconfig_t *msgn = (ctrlmsg_system_reconfig_t*) msg;
+
+	// Create reconfigure struct
+	ga_ioctl_reconfigure_t reconf;
+	bzero(&reconf, sizeof(reconf));
+
+	// Copy values from msg to msgn
+	reconf.id = msgn->reconfId;
+	reconf.crf = msgn->crf;
+	reconf.framerate_n = msgn->framerate;
+	reconf.framerate_d = 1;
+	reconf.bitrateKbps = msgn->bitrate;
+	reconf.width = msgn->width;
+	reconf.height = msgn->height;
+
+
+	// encoder
+	if(m_vencoder->ioctl) {
+		int err = m_vencoder->ioctl(GA_IOCTL_RECONFIGURE, sizeof(reconf), &reconf);
+		if(err < 0) {
+			ga_error("reconfigure encoder failed, err = %d.\n", err);
+		} else {
+			ga_error("reconfigure encoder OK, bitrate=%d; bufsize=%d; framerate=%d/%d.\n",
+					reconf.bitrateKbps, reconf.bufsize,
+					reconf.framerate_n, reconf.framerate_d);
+		}
+	}
+
+	return;
+}
+
+void
+handle_rttserver(ctrlmsg_system_t *msg){
+	pthread_t rttserverthread;
+
+	// Only start the handler once, in case multiple ctrlmsg signals are sent.
+	if(!rttThreadStarted){
+		rttThreadStarted = true;
+		pthread_mutex_init(&ping_queue_mutex, NULL);
+		ga_error("handle_rttserver: Initializing RTT response thread.\n");
+		if(pthread_create(&rttserverthread, NULL, rtt_handler, NULL) != 0){
+			ga_error("Cannot create UDP Handler thread.\n");
+			return;
+		}
+	}
+
+	return;
+}
+
+void
+handle_ping(ctrlmsg_system_t *msg) {
+	if(!rttThreadStarted){
+		return;
+	}
+	ctrlmsg_system_ping_t *msgn = (ctrlmsg_system_ping_t*) msg;
+
+	pthread_mutex_lock(&ping_queue_mutex);
+	ping_id_queue[ping_queue_size].rtt_id = msgn->ping_id;
```

Comment:
Instead of queueing these messages, we could just use the socket descriptor and send the replies from this function itself.